plugins/utils/wilder-nvim.nix

```diff
@@ -0,0 +1,80 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.wilder-nvim;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  boolToInt = value:
+    if value == null
+    then null
+    else if value
+    then "1"
+    else "0";
+in {
+  options.plugins.wilder-nvim =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "wilder-nvim";
+
+      package = helpers.mkPackageOption "wilder-nvim" pkgs.vimPlugins.wilder-nvim;
+
+      modes = helpers.defaultNullOpts.mkNullable (types.listOf types.str) "['/', '?']" ''
+        List of modes which wilderw will be active in.
+        Possible elements: '/', '?' and ':'
+      '';
+
+      enableCmdlineEnter = helpers.defaultNullOpts.mkBool true ''
+        If true calls wilder#enable_cmdline_enter().
+        Creates a new |CmdlineEnter| autocmd to which will start wilder
+        when the cmdline is entered.
+      '';
+
+      wildcharm = helpers.defaultNullOpts.mkStr "&wildchar" ''
```

Comment:
Well this default would be complex to express in Nix, it can be left as-is.

Reply:
As far as I understand this should be the same as default right? This would just translate into lua string, which should be the same I think.

Reply:
This is a vimscript expression right ? In any case the `mkStr` will add `""` around so it can't be an arbitrary expression, you would need to do an `types.either (str helpers.rawType)` for that, but I'm not sure it's worth it. You can leave it as this.

Reply:
Right, didn't notice that it actually is vimscript, at this point I just assume everything is lua. To be honest, I couldn't really find what it does anyway.